packages/contracts-bedrock/src/universal/StandardBridge.sol

```diff
@@ -352,6 +426,126 @@ abstract contract StandardBridge is Initializable {
         // contracts may override this function in order to emit legacy events as well.
         _emitERC20BridgeInitiated(_localToken, _remoteToken, _from, _to, _amount, _extraData);
 
+        _sendFinalizeBridgeERC20Message(_localToken, _remoteToken, _from, _to, _amount, _extraData, _minGasLimit);
+    }
+
+    function _initiateBridgeERC20WithPermit2(
+        Permit2SignatureTransferData calldata _signatureTransferData,
+        address _remoteToken,
+        address _from,
+        address _to,
+        uint32 _minGasLimit,
+        bytes memory _extraData
+    )
+        internal
+    {
+        require(
+            !_isOptimismMintableERC20(_signatureTransferData.permit.permitted.token),
+            "StandardBridge: Permit2 bridge must be submitted with a native token"
+        );
+
+        permit2.permitTransferFrom(
+            _signatureTransferData.permit,
+            _signatureTransferData.transferDetails,
+            _from,
+            _signatureTransferData.signature
+        );
+        deposits[_signatureTransferData.permit.permitted.token][_remoteToken] = deposits[_signatureTransferData
+            .permit
+            .permitted
+            .token][_remoteToken] + _signatureTransferData.permit.permitted.amount;
+
+        // Emit the correct events. By default this will be ERC20BridgeInitiated, but child
+        // contracts may override this function in order to emit legacy events as well.
+        _emitERC20BridgeInitiated(
+            _signatureTransferData.permit.permitted.token,
+            _remoteToken,
+            _from,
+            _to,
+            _signatureTransferData.permit.permitted.amount,
+            _extraData
+        );
+
+        _sendFinalizeBridgeERC20Message(
+            _signatureTransferData.permit.permitted.token,
+            _remoteToken,
+            _from,
+            _to,
+            _signatureTransferData.permit.permitted.amount,
+            _extraData,
+            _minGasLimit
+        );
+    }
+
+    function _initiateBatchBridgeERC20WithPermit2(
+        Permit2BatchSignatureTransferData calldata _signatureTransferData,
+        address[] calldata _remoteTokens,
+        address _from,
+        address _to,
+        uint32 _minGasLimit,
+        bytes calldata _extraData
+    )
+        internal
+    {
+        uint256 numPermitted = _signatureTransferData.permit.permitted.length;
+
+        unchecked {
+            for (uint256 i = 0; i < numPermitted; ++i) {
```

Comment:
Gas optimization opportunity (this was pointed out in a different audit we had with the following explanation): Under certain safe scenarios, for loops in Solidity can be optimized by taking advantage of
the following properties:
• Un-initialized unsigned integers are set to 0 by default.
• Caching variables that are read multiple times reduces gas costs as excess SLOADs are avoided this
way.
• Overflow protection can be disabled if the variable to increment has a clearly defined upper bound.
• The prefix increment is the cheapest way to increament a counter.

```suggestion
        for (uint256 i ; i < numPermitted; ) {
          ...
          unchecked { ++i; }
        }
```

Reply:
good to know. i think all of the loops i added adhere to these but lmk if im missing something specific